tests/pass/tree-borrows/propagation/nontrivial-structure.rs

```diff
@@ -1,44 +1,27 @@
 //@compile-flags: -Zmiri-tree-borrows -Zmiri-tag-gc=0
-#![feature(extern_types)]
-#[path = "../../../utils/mod.rs"]
+#[path = "../../../utils.rs"]
 mod utils;
-use utils::tb::num::*;
+use utils::*;
 
-fn main() {
-    unsafe {
-        let x = root!();
-        name!(x);
-        let xa = child!(x => &);
-        name!(xa);
-        let xb = child!(x => &);
-        name!(xb);
-        let xc = child!(x => &);
-        name!(xc);
-        let xaa = child!(xa => &);
-        name!(xaa);
-        let xab = child!(xa => &);
-        name!(xab);
-        let xba = child!(xb => &);
-        name!(xba);
-        let xbaa = child!(xba => &);
-        name!(xbaa);
-        let xbaaa = child!(xbaa => &);
-        name!(xbaaa);
-        let xbaaaa = child!(xbaaa => &);
-        name!(xbaaaa);
-        let xca = child!(xc => &);
-        name!(xca);
-        let xcb = child!(xc => &);
-        name!(xcb);
-        let xcaa = child!(xca => &);
-        name!(xcaa);
-        let xcab = child!(xca => &);
-        name!(xcab);
-        let xcba = child!(xcb => &);
-        name!(xcba);
-        let xcbb = child!(xcb => &);
-        name!(xcbb);
-        let alloc_id = alloc_id!(x);
-        fmt_tree!(alloc_id);
-    }
-}
+// Pushing the limits of the formating function that prints the tree structure
```

Comment:
Does this test formatting or propagataion (the latter is indicated by the folder name)?

Sounds like this could be merged with formatting.rs.